dbt_server/views.py

```diff
@@ -349,46 +386,59 @@ async def post_invocation(args: PostInvocationRequest):
+    """Gets invocation entity by `task_id`."""
+    invocation = (
+        convert_celery_result_to_invocation(
+            AbortableAsyncResult(task_id, app=celery_app)
+        )
+        if _lookup_abortable_async_result(task_id)
+        else get_not_found_invocation(task_id)
+    )
 
+    return JSONResponse(status_code=200, content=invocation.dict(exclude_unset=True))
 
-class GetInvocationResponse(BaseModel):
-    # Unique task id of invocation, same as request.
-    task_id: str
-    # Task state, it's one of celery worker state:
-    # - PENDING: The task is pending to be executed or never been created yet.
-    #   Due to celery backend design, we can't tell if task is pending or never
-    #   been created yet. Definitely there is a workaround, but we don't support
-    #   right now.
-    # - STARTED: The task is started by worker.
-    # - SUCCESS: The task is finished by worker.
-    # - FAILURE: The task is failed.
-    # - ABORTED: The task is aborted by user.
-    state: str
-    # Only exists if state = FAILURE, python exception type name that causes
-    # task is killed, e.g. dbtUsageException, WorkerLostError.
-    exc_type: Optional[str]
-    # Only exists if state = FAILURE, similar to exc_type, it includes error
-    # message of exception.
-    exc_message: Optional[str]
-
-
-@app.get("/invocation")
-async def get_invocation(args: GetInvocationRequest):
-    """Gets task state and other metadata."""
-    # TODO: Implement.
-    return JSONResponse(status_code=200, content={})
 
+class ListInvocationResponse(BaseModel):
+    # List of all invocations.
+    invocations: List[Invocation]
 
-class AbortInvocationRequest(BaseModel):
-    # Unique task id of invocation to be aborted.
-    task_id: str
+
+@app.get("/invocations")
+async def list_invocation():
+    """Gets invocation entity by `task_id`."""
+    return JSONResponse(
+        status_code=200,
+        content=ListInvocationResponse(
+            invocations=[
+                convert_celery_result_to_invocation(
+                    AbortableAsyncResult(task_id, app=celery_app)
+                )
+                for task_id in _list_all_task_ids()
+            ]
+        ).dict(exclude_unset=True),
+    )
 
 
-@app.post("/invocation/abort")
-async def abort_invocation(args: AbortInvocationRequest):
-    """Aborts tasks. Notice it's best effort, task may still finish or fail."""
-    # TODO: Implement.
-    return JSONResponse(status_code=200, content={})
+@app.post("/invocation/{task_id}/abort")
+async def abort_invocation(task_id: str):
+    """Aborts tasks. Notice it's best effort, task may still finish or fail.
+    Returns invocation model."""
+    if not _lookup_abortable_async_result(task_id):
+        return JSONResponse(
+            status_code=200,
+            content=get_not_found_invocation(task_id).dict(exclude_unset=True),
+        )
+
+    task = AbortableAsyncResult(task_id, app=celery_app)
+    if task.state in UNREADY_STATES:
```

Comment:
`UNREADY_STATES` seems like kind of a misleading term defined by celery-- could we maybe just add a comment to clarify that this includes `PENDING, RECEIVED, STARTED, REJECTED, RETRY`, or that `READY_STATES` are just `SUCCESS, FAILURE, REVOKED`? "Unready" sort of sounds like it won't cancel if a task has started.

Reply:
Makes sense to me, thank you!